server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -377,13 +377,21 @@ public interface IndexCommitListener {
          * {@link IndexCommitRef} prevents the {@link IndexCommitRef} files to be deleted from disk until the reference is closed. As such,
          * the listener must close the reference as soon as it is done with it.
          *
-         * @param shardId         the {@link ShardId} of shard
-         * @param store           the index shard store
-         * @param primaryTerm     the shard's primary term value
-         * @param indexCommitRef  a reference on the newly created index commit
-         * @param additionalFiles the set of filenames that are added by the new commit
+         * @param shardId            the {@link ShardId} of shard
+         * @param store              the index shard store
+         * @param primaryTerm        the shard's primary term value
+         * @param indexCommitRef     a reference on the newly created index commit
+         * @param additionalFiles    the set of filenames that are added by the new commit
+         * @param extraTransientData extra transient data relevant to the new commit
          */
-        void onNewCommit(ShardId shardId, Store store, long primaryTerm, IndexCommitRef indexCommitRef, Set<String> additionalFiles);
+        void onNewCommit(
+            ShardId shardId,
+            Store store,
+            long primaryTerm,
+            IndexCommitRef indexCommitRef,
+            Set<String> additionalFiles,
+            Map<String, String> extraTransientData
```

Comment:
I wonder if we should just pass Object here and let the subclass handle what types it wants?

Also,  perhaps we can call this "enginePreCommitData", i.e., data extracted prior to the commit by the engine.